pkg/primitives/vm.go

```diff
@@ -280,6 +282,29 @@ func (p *Primitives) virtualMachineProvisionImpl(ctx context.Context, wl *gridty
 		if err := p.vmMounts(ctx, deployment, config.Mounts, true, &machine); err != nil {
 			return result, err
 		}
+		if config.Corex {
+			rootBinPath := filepath.Join(mnt, "bin")
+			_, err = os.Stat(rootBinPath)
+			if err != nil {
+				return result, errors.Wrap(err, "Couldn't find rootfs")
+			}
+			corexPath := "/usr/bin/corex"
```

Comment:
Can the logic for the `copy` be moved to a new function? the logic should be eventually
```go
if config.Corex {
    if err := copy(<src>, filepath.Join(mnt, "corex"); err != nil {
       return errors.Wrap(err, "failed to inject corex binary")
   }
    entrypoint = "/bin/corex --ipv6 -d 7 --interface eth1"
}
```